common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResultFactory.java

```diff
@@ -0,0 +1,132 @@
+package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationErrorResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+public class AzureActiveDirectoryAuthorizationResultFactory extends AuthorizationResultFactory {
+
+    private static final String TAG = AzureActiveDirectoryAuthorizationResultFactory.class.getSimpleName();
+
+    private static final String ERROR_CODES = "error_codes";
+
+    @Override
+    public AuthorizationResult createAuthorizationResult(final int resultCode, final Intent data) {
```

Comment:
This is looking better -- can we lower the number of `return` statements?

Consider the following [contrived] example of adding logging to properties of our results -- multiple return statements increases our work, adding 4 new lines to the method for each `case`, instead of two.

Having a single return statement also makes it easier to set a breakpoint at the result and see what's coming out of a method

```java
public Fruit getProduce(Color color) {
    final Fruit result;

    switch(color) {
        case RED:
            result = new Apple();
            Log.info("Result weight: " + result.getWeight());
            Log.info("Result unit price: " + result.getPrice(Currency.US_DOLLARS));
            return result;

        case GREEN:
            result = new Pear();
            Log.info("Result weight: " + result.getWeight());
            Log.info("Result unit price: " + result.getPrice(Currency.US_DOLLARS));
            return result;
            
        default:
            throw new UnsupportedFruitException("Not an apple or a pear! Sad face!");
    }
}
```

versus
```java
public Fruit getProduce(Color color) {
    final Fruit result;

    switch(color) {
        case RED:
            result = new Apple();
            break;

        case GREEN:
            result = new Pear();
            break;

        default:
            throw new UnsupportedFruitException("Not an apple or a pear! Sad face!");
    }
    
    Log.info("Result weight: " + result.getWeight());
    Log.info("Result unit price: " + result.getPrice(Currency.US_DOLLARS));

    return result;
}
```

Reply:
I like idea definitely see this super useful for debugging. Thanks for the suggestion. Made required changes.